core_backend/app/tags/models.py

```diff
@@ -0,0 +1,158 @@
+from datetime import datetime
+from typing import List, Optional
+
+from sqlalchemy import (
+    Column,
+    DateTime,
+    ForeignKey,
+    Integer,
+    String,
+    Table,
+    delete,
+    select,
+)
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+
+from ..models import Base
+from .schemas import TagCreate
+
+content_tags_table = Table(
+    "content_tags",
+    Base.metadata,
+    Column("content_id", Integer, ForeignKey("content.content_id"), primary_key=True),
+    Column("tag_id", Integer, ForeignKey("tag.tag_id"), primary_key=True),
+)
+
+
+class TagDB(Base):
+    __tablename__ = "tag"
+
+    tag_id: Mapped[int] = mapped_column(Integer, primary_key=True, nullable=False)
+    tag_name: Mapped[str] = mapped_column(String(length=50), nullable=False)
+    created_datetime_utc: Mapped[datetime] = mapped_column(DateTime, nullable=False)
+    updated_datetime_utc: Mapped[datetime] = mapped_column(DateTime, nullable=False)
+    contents = relationship(
+        "ContentDB", secondary=content_tags_table, back_populates="content_tags"
+    )
+    user_id: Mapped[int] = mapped_column(
+        Integer, ForeignKey("user.user_id"), nullable=False
+    )
+
+    def __repr__(self) -> str:
+        return f"TagDB(tag_id={self.tag_id}, " f"tag_name='{self.tag_name}')>"
+
+
+async def save_tag_to_db(
+    user_id: int,
+    tag: TagCreate,
+    asession: AsyncSession,
+) -> TagDB:
+    """
+    Saves a tag in the database
+    """
+
+    tag_db = TagDB(
+        tag_name=tag.tag_name,
+        user_id=user_id,
+        created_datetime_utc=datetime.utcnow(),
+        updated_datetime_utc=datetime.utcnow(),
+    )
+
+    asession.add(tag_db)
+
+    await asession.commit()
+    await asession.refresh(tag_db)
+
+    return tag_db
+
+
+async def update_tag_in_db(
+    user_id: int,
+    tag_id: int,
+    tag: TagCreate,
+    asession: AsyncSession,
+) -> TagDB:
+    """
+    Updates a tag in the database
+    """
+
+    tag_db = TagDB(
+        tag_id=tag_id,
+        user_id=user_id,
+        tag_name=tag.tag_name,
+        created_datetime_utc=datetime.utcnow(),
+        updated_datetime_utc=datetime.utcnow(),
+    )
+
+    tag_db = await asession.merge(tag_db)
+    await asession.commit()
+    await asession.refresh(tag_db)
+
+    return tag_db
+
```

Comment:
Can we prevent the user from adding/updating a tag name to one that already exists in the database? I feel like not preventing this could lead to confusion for the user